server/src/main/java/org/opensearch/action/search/SearchUtils.java

```diff
@@ -40,4 +50,52 @@ public static StepListener<BiFunction<String, String, DiscoveryNode>> getConnect
         }
         return lookupListener;
     }
+
+    /**
+     * Delete list of pits, return success if all reader contexts are deleted ( or not found ).
+     */
+    public static void deletePits(
+        Collection<SearchContextIdForNode> contexts,
+        ActionListener<Integer> listener,
+        ClusterState state,
+        SearchTransportService searchTransportService
+    ) {
+        final Set<String> clusters = contexts.stream()
+            .filter(ctx -> Strings.isEmpty(ctx.getClusterAlias()) == false)
+            .map(SearchContextIdForNode::getClusterAlias)
+            .collect(Collectors.toSet());
+        StepListener<BiFunction<String, String, DiscoveryNode>> lookupListener = getConnectionLookupListener(
+            searchTransportService.getRemoteClusterService(),
+            state,
+            clusters
+        );
+        lookupListener.whenComplete(nodeLookup -> {
+            final GroupedActionListener<Boolean> groupedListener = new GroupedActionListener<>(
+                ActionListener.delegateFailure(
+                    listener,
+                    (l, result) -> l.onResponse(Math.toIntExact(result.stream().filter(r -> r).count()))
+                ),
+                contexts.size()
+            );
+
+            for (SearchContextIdForNode contextId : contexts) {
```

Comment:
this is inefficient. Can we club all requests to be sent to a single node and implement a transport layer method 

`freeContexts(List<ContextId> ids)`. that way number of transport calls made <= number of nodes(across clusters)

Reply:
we'll need a bulk delete or a bulk operation funda here to optimize.